HPCC-15545 ESDL Tool and WsESDLConfig to make ESDL Serv name optional

rpastrana opened the pull request:
- ESDL Service name is now optional when publishing an ESDL definition
  If defintion only contains one service definition

- ESDL Service name is now optional when publishing an ESDL Binding
  If defintion only contains one service definition- Removes extra space in placeholder ip list text

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [x] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@wangkx please review

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 26b8abf131ecaef6596f61af85f967b17e60bec3
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 94.338 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |9 |9 |0 |
|setup (thor) |9 |9 |0 |
|setup (roxie) |9 |9 |0 |
|test (hthor) |720 |720 |0 |
|test (thor) |615 |615 |0 |
|test (roxie) |745 |745 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -300,25 +300,29 @@ bool CWsESDLConfigEx::existsESDLMethodDef(const char * esdlDefinitionName, unsig
                         if (stricmp(pChildNode->queryProp("@name"), methodName)==0)
                         {
+                            if (!esdlServiceName.length())
+                                esdlServiceName.set(pCurrService->queryProp("@name"));
```
What if esdlServiceName has some value which != pCurrService->queryProp("@name")?

rpastrana replied:
I think the conditional at 309 prevents that scenario; we only enter this block if either esdlServiceName == @name, or esdlService is empty and there's only one service...

wangkx commented on tools/esdlcmd/esdl-publish.cpp:
```diff
@@ -236,8 +245,8 @@ class EsdlPublishCmd : public EsdlPublishCmdCommon
-        if (optESDLService.isEmpty())
-            throw MakeStringException( 0, "Name of ESDL based service must be provided" );
+        //if (optESDLService.isEmpty())
+        //    throw MakeStringException( 0, "Name of ESDL based service must be provided" );
```
Remove?

rpastrana replied:
Will remove

wangkx reviewed (commented):
@rpastrana 2 questions

rpastrana committed:
HPCC-15545 ESDL Tool and WsESDLConfig to make ESDL Serv name optional

- ESDL Service name is now optional when publishing an ESDL definition
  If defintion only contains one service definition

- ESDL Service name is now optional when publishing an ESDL Binding
  If defintion only contains one service definition- Removes extra space in placeholder ip list text

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx changes squashed, please review.

wangkx commented:
@rpastrana looks fine.

rpastrana commented:
@richardkchapman please merge

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 74803a7b4697f356761626703911bfa6d172429b
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.362 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |9 |9 |0 |
|setup (thor) |9 |9 |0 |
|setup (roxie) |9 |9 |0 |
|test (hthor) |720 |720 |0 |
|test (thor) |616 |616 |0 |
|test (roxie) |745 |745 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
